File: src/main/java/nz/co/searchwellington/commentfeeds/detectors/BlogspotCommentFeedDetector.java
package nz.co.searchwellington.commentfeeds.detectors;

import org.springframework.stereotype.Component;

import java.net.URL;

@Component
public class BlogspotCommentFeedDetector implements CommentFeedDetector {

    public boolean isValid(URL url) {
        return url != null && url.toExternalForm().contains("blogspot.com")
                && url.toExternalForm().contains("/comments/");
    }
    
}

File: src/main/scala/nz/co/searchwellington/commentfeeds/detectors/BlogspotCommentFeedDetector.scala
package nz.co.searchwellington.commentfeeds.detectors

import nz.co.searchwellington.model.Resource
import org.springframework.stereotype.Component

import java.net.URL

@Component
class BlogspotCommentFeedDetector extends CommentFeedDetector {
  override def isValid(url: URL, source: Resource): Boolean = {
    url.getHost.endsWith("blogspot.com") && url.getPath.contains("/comments/")
  }
}